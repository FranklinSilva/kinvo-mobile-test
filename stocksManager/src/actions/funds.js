import api from '../services/api';
import deviceStorage from '../services/deviceStorage';
import apiUtils from '../services/apiUtils';

export const getFunds = () =>{

    return async (dispatch) => {
        return api({
            method: 'get',
            url: "funds",
            headers: {
                "Accept": "application/json"
            },
            })  
            .then((response) => {
                dispatch(getFundsSuccess(response.data.data));
            })
            .catch((error) => {
                // Handle returned errors here
                apiUtils.handleError(error);
                dispatch();
            });
    }
}


export const getFundsSuccess = (funds) => {    
    return {
        type: 'GET_FUNDS',
        payload:{
            funds
        }
    }
}