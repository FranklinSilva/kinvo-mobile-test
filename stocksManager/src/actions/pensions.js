import api from '../services/api';
import deviceStorage from '../services/deviceStorage';
import apiUtils from '../services/apiUtils';

export const getPensions = () =>{

    return async (dispatch) => {
        return api({
            method: 'get',
            url: "pension",
            headers: {
                "Accept": "application/json"
            },
            })  
            .then((response) => {
                dispatch(getPensionsSuccess(response.data.data));
            })
            .catch((error) => {
                // Handle returned errors here
                //apiUtils.handleError(error);
                dispatch();
            });
    }
}


export const getPensionsSuccess = (pensions) => {    
    return {
        type: 'GET_PENSIONS',
        payload:{
            pensions
        }
    }
}