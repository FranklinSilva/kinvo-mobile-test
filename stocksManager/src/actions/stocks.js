import api from '../services/api';
import deviceStorage from '../services/deviceStorage';
import apiUtils from '../services/apiUtils';

export const getStocks = () =>{

    return async (dispatch) => {
        return api({
            method: 'get',
            url: "stocks",
            headers: {
                "Accept": "application/json"
            },
            })  
            .then((response) => {
                dispatch(getStocksSuccess(response.data.data));
            })
            .catch((error) => {
                // Handle returned errors here
                //apiUtils.handleError(error);
                dispatch();
            });
    }
}


export const getStocksSuccess = (stocks) => {    
    return {
        type: 'GET_STOCKS',
        payload:{
            stocks
        }
    }
}

export const setFavoritedStock = (stockId) =>{

    return async (dispatch) => {
        return dispatch(setFavoritedStockSucess(stockId));
    }
}

export const setFavoritedStockSucess = (stockId) => {
    
    return {
        type: 'SET_STOCK_TO_FAVORITE',
        payload:{
            stockId
        }
    }
}