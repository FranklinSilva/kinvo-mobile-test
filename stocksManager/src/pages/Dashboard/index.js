import React, { useState } from 'react';
import {View, Text, FlatList} from 'react-native';
import FrontCard from '../../components/frontCard';

const CATEGORIES = [
    {imageUrl:  require("../../images/icons/acoes.png"), route: "Stocks", title: "Ações", subtitle: "Nacionais", new: false},
    {imageUrl:  require("../../images/icons/funds.png"), route: "Funds", title: "Fundos", subtitle: "De Investimentos", new: true},
    {imageUrl:  require("../../images/icons/previdencia.png"), route: "Previdences", title: "Previdências", subtitle: "Privadas", new: false}
]

const Dashboard = (props) => {

  const renderItems = ({item, index}) => {
    return (
        <FrontCard index={index} category={item} />
        )
  }

  return (
    <View>
        <FlatList 
            data={CATEGORIES}
            keyExtractor={(item, index) => index + ""}
            renderItem={renderItems}
            inverted={false}
            initialNumToRender={3}
            horizontal={false}/>   
    </View>
  );
}

export default Dashboard;