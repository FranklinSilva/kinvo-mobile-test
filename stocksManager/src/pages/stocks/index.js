import React, { useState } from 'react';
import {View, Text, FlatList} from 'react-native';

const Stocks = (props) => {

  return (
    <View>
        <Text>Stocks</Text>
    </View>
  );
}

export default Stocks;