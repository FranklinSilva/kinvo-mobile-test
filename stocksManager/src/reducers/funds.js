import { compareStocks } from '../services/utils';


export default function funds(state=[], action) {

    switch (action.type) {
        case 'GET_FUNDS':
            let sorted = action.payload.funds;
            sorted.sort(compareStocks);
            return sorted
        default:
            return state;
    }
}