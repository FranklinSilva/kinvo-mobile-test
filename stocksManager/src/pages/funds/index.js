import React, { useState, useEffect } from 'react';
import {View, Text, FlatList} from 'react-native';
import FundsCard from '../../components/fundsCard.js'
import ApiError from '../../components/apiError';
import LoadingIndicator from '../../components/loadingIndicator';

import {connect} from 'react-redux';
import * as fundsActions from '../../actions/funds';
import { bindActionCreators } from 'redux';

const Funds = (props) => {
  const [loading, setLoading] = useState(false);
  const [fetchError, setFetchError] = useState(false);
  const [shouldCallApi, setShouldCallApi] = useState(false);

  useEffect(() => {
    if(props.funds.length == 0){
      setLoading(true)
      props.getFunds()
          .then(() => {
            setFetchError(false);

            if(props.funds.length == 0){
              setShouldCallApi(!shouldCallApi);
              setLoading(false)
            }
          })
          .catch(() => {
              setLoading(false)
              setFetchError(true)
          })
    }
  },[shouldCallApi]);

  const renderItems = ({item, index}) => {
    return (
        <FundsCard index={index} fund={item} />
        )
  }


  return (
    <View>
        {fetchError == true &&
          <ApiError callTryAgain={() => setShouldCallApi(!shouldCallApi)} />
        }
        
        {!fetchError && loading &&
          <LoadingIndicator/>
        }
        {!fetchError && !loading &&
        <FlatList 
            data={props.funds}
            keyExtractor={(item, index) => index + ""}
            renderItem={renderItems}
            inverted={false}
            initialNumToRender={3}
            horizontal={false}/>
        }   
    </View>
  );
}

const mapStateToProps = state => ({
  funds: state.funds,
})

const mapDispatchToProps = dispatch => 
bindActionCreators({...fundsActions}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Funds);