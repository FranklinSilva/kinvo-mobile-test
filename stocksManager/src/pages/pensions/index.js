import React, { useState, useEffect } from 'react';
import {View, Text, FlatList, TouchableOpacity} from 'react-native';
import PensionCard from '../../components/pensionCard.js'
import ApiError from '../../components/apiError';
import LoadingIndicator from '../../components/loadingIndicator';

//redux things
import {connect} from 'react-redux';
import * as pensionsActions from '../../actions/pensions';
import { bindActionCreators } from 'redux';
import styles from './styles.js';

const FILTERLIMITS = {noTaxFundsValue: 0, minimumFundsValue: 100, redemptionTermValue: 1}

const Pensions = (props) => {
  const [loading, setLoading] = useState(false);
  const [fetchError, setFetchError] = useState(false);
  const [pensions, setPension] = useState([]);
  const [shouldCallApi, setShouldCallApi] = useState(false);
  const [pensionsFilter, setPensionsFilter] = useState({noTax: false, min100: false, d1: false});

  useEffect(() => {
    if(pensions.length == 0){
      setLoading(true)
      props.getPensions()
          .then(() => {
            setPension(props.pensions)
            setFetchError(false)
            
            if(pensions.length == 0){
              setShouldCallApi(!shouldCallApi);
              setLoading(false)
            }

          })
          .catch(() => {
              setLoading(false)
              setFetchError(true)
          })
    }
  },[shouldCallApi]);

  useEffect(() => {
    let noTaxActive = pensionsFilter.noTax;
    let min100Active = pensionsFilter.min100;
    let d1Active = pensionsFilter.d1;
    
    let pensionsFiltered = props.pensions;
    
    if(noTaxActive)
      pensionsFiltered = pensionsFiltered.filter(e=> e.tax == FILTERLIMITS.noTaxFundsValue);

    if(min100Active)
      pensionsFiltered = pensionsFiltered.filter(e=> e.minimumValue == FILTERLIMITS.minimumFundsValue);

    if(d1Active)
      pensionsFiltered = pensionsFiltered.filter(e=> e.redemptionTerm == FILTERLIMITS.redemptionTermValue);
  
    setPension(pensionsFiltered);
    setPensionsFilter(pensionsFilter);

  }, [pensionsFilter]);

  const renderItems = ({item, index}) => {
    return (
        <PensionCard index={index} pension={item} />
        )
  }


  return (
    <View>
      {!fetchError &&
        <View>
          <View style={styles.pensionFilter}>
            <TouchableOpacity style={pensionsFilter.noTax ? styles.pensionFilterItemSelected : styles.pensionFilterItem} onPress={() => setPensionsFilter({...pensionsFilter, noTax: !pensionsFilter.noTax})}>
              <Text style={pensionsFilter.noTax ? styles.pensionFilterItemTextSelected : styles.pensionFilterItemText }>SEM TAXA</Text>
            </TouchableOpacity>
            <TouchableOpacity style={pensionsFilter.min100 ? styles.pensionFilterItemSelected : styles.pensionFilterItem} onPress={() => setPensionsFilter({...pensionsFilter, min100: !pensionsFilter.min100})}>
              <Text style={pensionsFilter.min100 ? styles.pensionFilterItemTextSelected : styles.pensionFilterItemText}>R$ 100,00</Text>
            </TouchableOpacity>
            <TouchableOpacity style={pensionsFilter.d1 ? styles.pensionFilterItemSelected : styles.pensionFilterItem} onPress={() => setPensionsFilter({...pensionsFilter, d1: !pensionsFilter.d1})}>
              <Text style={pensionsFilter.d1 ? styles.pensionFilterItemTextSelected : styles.pensionFilterItemText}>D+1</Text>
            </TouchableOpacity>
          </View>

          <View style={styles.line}></View>
        </View>
      }
      {pensions.length == 0 && !loading && !fetchError &&
        <View style={styles.pensionFilter}>
          <Text style={styles.notFoundFilter}>Nenhum resultado foi encontrado para os filtros selecionados.</Text>
        </View>
      }

      {fetchError &&
        <ApiError callTryAgain={() => setShouldCallApi(!shouldCallApi)} />
      }

      {!fetchError && loading &&
        <LoadingIndicator/>
      }

      {!fetchError && !loading &&
        <FlatList 
          data={pensions}
          keyExtractor={(item, index) => index + ""}
          renderItem={renderItems}
          horizontal={false}
          style={{height: '85%'}}/>
      }   
    </View>
  );
}

const mapStateToProps = state => ({
  pensions: state.pensions,
})

const mapDispatchToProps = dispatch => 
bindActionCreators({...pensionsActions}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Pensions);