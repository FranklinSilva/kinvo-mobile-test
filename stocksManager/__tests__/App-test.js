/**
 * @format
 */

 import 'react-native';
 import React from 'react';
 import Dashboard from '../src/pages/dashboard';
 import Stocks from '../src/pages/stocks';
 import Funds from '../src/pages/funds';
 import Pensions from '../src/pages/pensions';
 import { createStore, applyMiddleware } from 'redux';
 import thunk from 'redux-thunk';
 import { Provider } from 'react-redux';
 import reducer from '../src/reducers';
 import {getStocks, getStocksSuccess} from '../src/actions/stocks';
 import {getFunds, getFundsSuccess} from '../src/actions/funds';
 import {getPensions, getPensionsSuccess} from '../src/actions/pensions';
 
 // Note: test renderer must be required after react-native.
 import renderer, { create, act } from 'react-test-renderer';
 
 jest.useFakeTimers();


 //Mocks
 jest.mock('@react-navigation/core')

 jest.mock('@react-native-community/async-storage', () => {
  return {
    getItem: async (...args) => args,
    setItem: async (...args) => args,
    removeItem: async (...args) => args,
  };
});

 
 jest.mock('react-navigation', () => ({
   withNavigation: (Component) => (props) => (
     <Component navigation={{ navigate: jest.fn() }} {...props} />
   ),
 }))

 jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: ''
}))
 
 const store = createStore(reducer, applyMiddleware(thunk))
 
 
 /* DASHBOARD PAGE */
 const dashboard = create(
 <Provider store={store}>
   <Dashboard />
 </Provider>);
 
 it('dashboard renders correctly', () => {
   expect(dashboard).toMatchSnapshot();
 });
 
 /* STOCKS PAGE */
 const stocks = create(
   <Provider store={store}>
     <Stocks />
   </Provider>);
  it('Get stocks', () => {
    expect(getStocks()).toBeDefined();
    expect(getStocksSuccess({})).toBeDefined();
  })
 
 it('stocks renders correctly', () => {
   expect(stocks).toMatchSnapshot();
 });


  /* FUNDS PAGE */
  const funds = create(
    <Provider store={store}>
      <Funds />
    </Provider>);
   it('Get funds', () => {
     expect(getFunds()).toBeDefined();
     expect(getFundsSuccess({})).toBeDefined();
   })
  
  it('funds renders correctly', () => {
    expect(funds).toMatchSnapshot();
  });


  /* PENSIONS PAGE */
  const pensions = create(
    <Provider store={store}>
      <Pensions />
    </Provider>);
    it('Get pensions', () => {
      expect(getPensions()).toBeDefined();
      expect(getPensionsSuccess({})).toBeDefined();
    })
  
  it('pensions renders correctly', () => {
    expect(pensions).toMatchSnapshot();
  });