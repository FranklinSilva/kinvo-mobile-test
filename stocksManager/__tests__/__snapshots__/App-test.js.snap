// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard renders correctly 1`] = `
<View>
  <RCTScrollView
    data={
      Array [
        Object {
          "imageUrl": Object {
            "testUri": "../../../src/images/icons/acoes.png",
          },
          "new": false,
          "route": "Stocks",
          "subtitle": "Nacionais",
          "testID": "stocksBtn",
          "title": "Ações",
        },
        Object {
          "imageUrl": Object {
            "testUri": "../../../src/images/icons/funds.png",
          },
          "new": true,
          "route": "Funds",
          "subtitle": "De Investimentos",
          "testID": "fundsBtn",
          "title": "Fundos",
        },
        Object {
          "imageUrl": Object {
            "testUri": "../../../src/images/icons/previdencia.png",
          },
          "new": false,
          "route": "Pensions",
          "subtitle": "Privadas",
          "testID": "pensionsBtn",
          "title": "Previdências",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={3}
    invertStickyHeaders={false}
    inverted={false}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#707B811A",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginHorizontal": 25,
              "marginTop": 20,
              "opacity": 1,
              "paddingHorizontal": 20,
              "paddingVertical": 32,
            }
          }
          testID="stocksBtn"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#ECF0F2",
                  "borderRadius": 30,
                  "padding": 10,
                }
              }
            >
              <Image
                resizeMode="contain"
                source={
                  Object {
                    "testUri": "../../../src/images/icons/acoes.png",
                  }
                }
                style={
                  Object {
                    "height": 25,
                    "width": 25,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "marginLeft": 12,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#6F4DBF",
                    "fontSize": 16,
                    "fontWeight": "bold",
                  }
                }
              >
                Ações
              </Text>
              <Text
                style={
                  Object {
                    "color": "#627179",
                    "fontSize": 10,
                    "fontWeight": "bold",
                  }
                }
              >
                Nacionais
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#707B811A",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginHorizontal": 25,
              "marginTop": 20,
              "opacity": 1,
              "paddingHorizontal": 20,
              "paddingVertical": 32,
            }
          }
          testID="fundsBtn"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#ECF0F2",
                  "borderRadius": 30,
                  "padding": 10,
                }
              }
            >
              <Image
                resizeMode="contain"
                source={
                  Object {
                    "testUri": "../../../src/images/icons/funds.png",
                  }
                }
                style={
                  Object {
                    "height": 25,
                    "width": 25,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "marginLeft": 12,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#6F4DBF",
                    "fontSize": 16,
                    "fontWeight": "bold",
                  }
                }
              >
                Fundos
              </Text>
              <Text
                style={
                  Object {
                    "color": "#627179",
                    "fontSize": 10,
                    "fontWeight": "bold",
                  }
                }
              >
                De Investimentos
              </Text>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#40C5D6",
                  "borderRadius": 20,
                  "padding": 5,
                  "width": 70,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 10,
                    "textAlign": "center",
                  }
                }
              >
                Novo
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#707B811A",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginHorizontal": 25,
              "marginTop": 20,
              "opacity": 1,
              "paddingHorizontal": 20,
              "paddingVertical": 32,
            }
          }
          testID="pensionsBtn"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#ECF0F2",
                  "borderRadius": 30,
                  "padding": 10,
                }
              }
            >
              <Image
                resizeMode="contain"
                source={
                  Object {
                    "testUri": "../../../src/images/icons/previdencia.png",
                  }
                }
                style={
                  Object {
                    "height": 25,
                    "width": 25,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "marginLeft": 12,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#6F4DBF",
                    "fontSize": 16,
                    "fontWeight": "bold",
                  }
                }
              >
                Previdências
              </Text>
              <Text
                style={
                  Object {
                    "color": "#627179",
                    "fontSize": 10,
                    "fontWeight": "bold",
                  }
                }
              >
                Privadas
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`funds renders correctly 1`] = `
<View>
  <ActivityIndicator
    animating={true}
    color="#6F4DBF"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "color": "#6F4DBF",
        "left": 355,
        "position": "relative",
        "right": 0,
        "top": 375,
        "width": 40,
      }
    }
  />
</View>
`;

exports[`pensions renders correctly 1`] = `
<View>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "marginHorizontal": 10,
          "marginTop": 20,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 15,
            "opacity": 1,
            "paddingHorizontal": 25,
            "paddingVertical": 7,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#627179",
              "fontSize": 12,
              "fontWeight": "bold",
            }
          }
        >
          SEM TAXA
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 15,
            "opacity": 1,
            "paddingHorizontal": 25,
            "paddingVertical": 7,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#627179",
              "fontSize": 12,
              "fontWeight": "bold",
            }
          }
        >
          R$ 100,00
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 15,
            "opacity": 1,
            "paddingHorizontal": 25,
            "paddingVertical": 7,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#627179",
              "fontSize": 12,
              "fontWeight": "bold",
            }
          }
        >
          D+1
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#707B811A",
          "flexDirection": "row",
          "height": 2,
          "marginHorizontal": 30,
          "marginTop": 15,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "marginHorizontal": 10,
        "marginTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#627179",
          "fontSize": 12,
          "fontWeight": "bold",
          "marginHorizontal": 90,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      Nenhum resultado foi encontrado para os filtros selecionados.
    </Text>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "height": "85%",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`stocks renders correctly 1`] = `
<View>
  <ActivityIndicator
    animating={true}
    color="#6F4DBF"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "color": "#6F4DBF",
        "left": 355,
        "position": "relative",
        "right": 0,
        "top": 375,
        "width": 40,
      }
    }
  />
</View>
`;
